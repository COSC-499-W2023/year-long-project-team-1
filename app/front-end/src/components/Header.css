/*
 * Created on Sun Oct 29 2023
 * Author: Connor Doman
 */

.site-header {
    background-color: var(--privacy-pal-primary-color);
    text-align: center;
    height: 25vh;
    overflow: hidden;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.site-header .logo {
    max-width: 6rem;
    height: auto;
}

.site-header .text {
    font-size: 30px;
    font-weight: bolder;
    color: white;
}
