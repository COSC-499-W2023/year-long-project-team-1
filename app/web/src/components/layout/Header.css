/*
 * Created on Sun Oct 29 2023
 * Author: Connor Doman
 */

.site-header {
    background-color: var(--pf-v5-global--primary-color--100);
    text-align: center;
    height: 25vh;
    /* overflow: hidden; */
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.site-header .logo {
    max-width: 6rem;
    height: auto;
}

.site-header .text {
    font-size: 30px;
    font-weight: bolder;
    color: white;
}
