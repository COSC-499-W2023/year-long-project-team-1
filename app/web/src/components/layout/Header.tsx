/*
 * Copyright [2023] [Privacypal Authors]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

"use server";
import Image from "next/image";
import logo from "@assets/logo.png";
import Link from "next/link";
import { LoginLogout } from "@components/auth/link/LoginLogout";
import { getAuthSession } from "@app/actions";

const style = {
  header: {
    backgroundColor: "var(--pf-v5-global--primary-color--100)",
    textAlign: "center",
    height: "fit-content",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    padding: "0",
    position: "sticky",
    top: "0",
    zIndex: "1000",
  } as React.CSSProperties,

  logo: {
    maxWidth: "6rem",
    height: "auto",
  },
  text: {
    fontSize: "30px",
    fontWeight: "bolder",
    color: "white",
    margin: "0.5rem 0",
  },
  loginLinks: {
    width: "100%",
    padding: "0.5rem 1rem",
    display: "flex",
    justifyContent: "flex-end",
    backgroundColor: "var(--pf-v5-global--primary-color--100)",
    gap: "1rem",
  },
  link: {
    color: "white",
    textDecoration: "underline",
  },
};

export const Header = async () => {
  const user = await getAuthSession();

  return (
    <header style={style.header}>
      <div style={style.loginLinks}>
        <LoginLogout user={user} style={style.link} />
      </div>
      <Link href="/">
        <Image alt="logo" style={style.logo} src={logo} />
      </Link>
      <div>
        <h1 style={style.text}>
          A SOLUTION TO ABSOLUTE{" "}
          <span style={{ color: "#F58658" }}>PRIVACY.</span>
        </h1>
      </div>
    </header>
  );
};
