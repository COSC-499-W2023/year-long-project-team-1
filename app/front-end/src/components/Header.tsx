/*
 * Created on Sun Oct 29 2023
 * Author: Connor Doman
 */

import "./Header.css";
import Image from "next/image";
import logo from "@assets/logo.png";

export const Header = () => {
    return (
        <header className="site-header">
            <Image alt="logo" className="logo" src={logo} />
            <div>
                <h1 className="text">
                    A SOLUTION TO ABSOLUTE <span style={{ color: "#F58658" }}>PRIVACY.</span>
                </h1>
            </div>
        </header>
    );
};
