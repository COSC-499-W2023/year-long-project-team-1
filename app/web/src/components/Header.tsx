/*
 * Created on Sun Oct 29 2023
 * Author: Connor Doman
 */

import "./Header.css";
import Image from "next/image";
import logo from "@assets/logo.png";
import Link from "next/link";

export const Header = () => {
    return (
        <header className="site-header">
            <Link href="/">
                <Image alt="logo" className="logo" src={logo} />
            </Link>
            <div>
                <h1 className="text">
                    A SOLUTION TO ABSOLUTE <span style={{ color: "#F58658" }}>PRIVACY.</span>
                </h1>
            </div>
        </header>
    );
};
