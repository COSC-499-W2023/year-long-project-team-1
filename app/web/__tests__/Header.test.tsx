/*
 * Created on Mon Oct 30 2023
 * Author: Connor Doman
 */

import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Header } from "@components/Header";

describe("Header", () => {
    it("matches snapshot", () => {
        const wrapper = render(<Header />);
        expect(wrapper).toMatchSnapshot();
    });

    it("contains an HTML header element", () => {
        render(<Header />);
        expect(screen.getByRole("banner")).toBeInTheDocument();
    });

    it("renders the header", () => {
        render(<Header />);
        expect(screen.getByRole("heading", { name: "A SOLUTION TO ABSOLUTE PRIVACY." })).toBeInTheDocument();
    });

    it("renders the header image", () => {
        render(<Header />);
        expect(screen.getByRole("img", { name: "logo" })).toBeInTheDocument();
    });
});
