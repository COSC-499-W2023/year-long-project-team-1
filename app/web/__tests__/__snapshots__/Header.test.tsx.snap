// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="site-header"
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            class="logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
        <div>
          <h1
            class="text"
          >
            A SOLUTION TO ABSOLUTE 
            <span
              style="color: rgb(245, 134, 88);"
            >
              PRIVACY.
            </span>
          </h1>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="site-header"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          class="logo"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </a>
      <div>
        <h1
          class="text"
        >
          A SOLUTION TO ABSOLUTE 
          <span
            style="color: rgb(245, 134, 88);"
          >
            PRIVACY.
          </span>
        </h1>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
